src/manifest.rs

```diff
@@ -95,6 +111,42 @@ impl Default for OciImageManifest {
     }
 }
 
+impl From<OciImageIndex> for OciManifest {
+    fn from(m: OciImageIndex) -> Self {
+        Self::ImageIndex(m)
+    }
+}
+impl From<OciImageManifest> for OciManifest {
+    fn from(m: OciImageManifest) -> Self {
+        Self::Image(m)
+    }
+}
+
+fn write_json(obj: &impl Serialize, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+    write!(
+        f,
+        "{}",
+        serde_json::to_string(obj).map_err(|_| std::fmt::Error)?
+    )
+}
+impl std::fmt::Display for OciManifest {
```

Comment:
Serializing JSON seems pretty heavy for using in a `Display` implementation, particularly `Display` is used in a lot of logging contexts which will incur a decent performance penalty each time it is logged. Perhaps this should be our custom debug implementation or we should drop it.

@flavio any thoughts here (since I think you use this library even more than I do)?

Reply:
I understand the concern. Since JSON is how most people interact with manifests, I figured having that as the `Display` impl seemed appropriate. Debug output is still available for those who want that usage.

Reply:
[According to the docs](https://doc.rust-lang.org/std/fmt/trait.Display.html), `Display` is meant for user-facing output so I don't see an issue on preformance if we render the object as JSON.

That being said, I do have concerns over the ergonomics of this API. What if the caller wanted to display the object as YAML? What if the user wanted it to be in a "prettified" format? I could see others asking for an `output_format` field on the `OciManifest` to modify `Display`'s output... I don't think that's ideal.

I think we should follow the "format" defined in https://doc.rust-lang.org/rust-by-example/hello/print/print_display.html. In other words:

```
 fmt::Display for OciManifest {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        // display as "image: {}, manifest: {}"
        // ....
    }
}
```

Reply:
@thomastaylor312  I rely a lot on `tracing` to handle my log messages. With `tracing` I can decided how an object is going to be handled. You could use the `Display`, the `Debug` or just do something ad-hoc. Hence, performance hits inside of logs are not something concerning me.

@bacongobbler I agree with @jsoverson, being the manifests JSON objects, it felt natural to have `Display` return a prettified JSON.

Having said that, I would be in favor of providing a more "standard" behaviour, like the one described above by @bacongobbler.

Users can then rely on `serde` or other libraries to pretty-print the manifest in their favourite way

Reply:
Ok, so to summarize, I think we use the more standard way of implementing `Display` and leave it up to consumers of the crate to print how they want. If we find that most users are copy/pasting the same code in the future, we can always expose that as part of the crate

Reply:
that sounds good to me @thomastaylor312

Reply:
This PR has been pending since a long time, I would like to get this merged so that we can tag a new release of oci-distribution.

I've tried to implement the change requested, but I realized this would be pretty painful: we do have a lots of attributes inside of these Structs and the other Structs that belong to them.
Some of these attributes are also `Optional<>`, hence they need to be handled with some "care" inside of `Display`.

There's an abundance of attributes inside of all these structures. Writing all the Display implementation would take some time (and maintenance), plus I fear we might end up with an output that is just messy...

I changed my mind, I think we should go ahead with what the author proposes: use the json format inside of the `Display` implementations